onnx/onnx.in.proto

```diff
@@ -242,10 +370,38 @@ message ModelProto {
   optional string doc_string = 6;
 
   // The parameterized graph that is evaluated to execute the model.
+  // This graph cannot include operators defined in ai.onnx.training.
   optional GraphProto graph = 7;
 
   // Named metadata values; keys should be distinct.
   repeated StringStringEntryProto metadata_props = 14;
+
+  // Training-specific information. Sequentially executing all stored
+  // `TrainingInfoProto.algorithm`s and assigning their outputs following
+  // the corresponding `TrainingInfoProto.update_binding`s is one training
+  // iteration. Similarly, to initialize the model
+  // (as if training hasn't happened), the user should sequentially execute
+  // all stored `TrainingInfoProto.initialization`s with the associated
+  // `TrainingInfoProto.initialization_binding`s.
+  //
+  // If this field is empty, the training behavior of the model is undefined.
+  repeated TrainingInfoProto training_info = 20;
+
+  // The following denote mutable global variables visible in ModelProto.graph as all as
+  // all graphs in TrainingInfoProto. These are initialized to the values specified
+  // in the initializer below, and then updated by initializer-bindings and
+  // update-bindings.
+  //
+  // If an input or initializer in a graph has the same name as a global variable,
+  // then the local variable in the graph will hide the global variable.
```

Comment:
Graph initializer means `declaring a variable so local variable hides outer-scope's variable`. Operator output means `declaring a variable if that variable doesn't exist; if that variable exists, it will be an assignment which we want to avoid`.